app/models/git_repository.rb

```diff
@@ -25,6 +25,20 @@ def self.delete_repo_dir(id, directory_name)
     FileUtils.rm_rf(directory_name)
   end
 
+  # Check the connection for the remote.
+  #
+  # Does not clone a repo or use an existing one, and instead initializes a
+  # repo in a temp path to check the remote to see if it can accept
+  # connections.
+  def check_connection
+    Dir.mktmpdir do |tmp_repo_dir|
+      repo_opts    = worktree_params.merge(:new => true, :path => tmp_repo_dir)
```

Comment:
`:new => true` here is akin to `git init`, so while there is going to be a bit of Disk IO as a result for every time we do this, it is far better then downloading the entire repo in question to check connectivity.

Alternatively to all of this, we might be able to find a way to "ping" the endpoint using the URL with just `net/http`, but this allows us to reuse a lot of the credential management in this class an `lib/git_worktree.rb`, so I think the cost of a bit of extraneous Disk IO is worth it in this case.

Reply:
When you do this with git proper, typically the best way is to call `git ls-remote`.  Wonder if there's an equivalent in rugged.

Reply:
Unfortunately, I didn't find anything that did that, otherwise I would have.

I wasn't sure if shelling out to do a `git ls-remote` was worth it, though, since we are getting some `segfaults`, maybe doing that instead is a better option.

However, this would need a bunch of special work to handle `proxy` and other things if we did a shell out.

Reply:
I did find this 🤷‍♂️ https://github.com/libgit2/rugged/blob/master/ext/rugged/rugged_remote.c#L736

Reply:
Agreed, I wouldn't shell out...I was just trying to see if we could avoid the Dir.mktmpdir, which git ls-remote is generally able to work without a clone/diretctory and an alagous method in rugged might be preferable.

Reply:
@Fryguy Still suffers from the fact that it requires a `remote` object, which can't exist on it's own.  As far as I can tell that is.

`.check_connection` seems to be the same thing, but just returns a `true`/`false`.

Reply:
For the record:  I am 100% in agreement that I wish the `Dir.mktmpdir` wasn't needed, and makes me a bit sad... 😞

Reply:
Yeah, I tried all kinds of hackery and can't get it to work... thanks for humoring me 😆